### PR TITLE
added ml to index.json

### PR DESCRIPTION
**Pull request by Mossaka**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yungezz** [comment]
hi @fengzhou-msft is this the release for installation fixing? thanks

File: src/index.json [hunk 1]
```diff
@@ -10312,6 +10312,51 @@
...
+                                "msrest (>=0.6.18)",
+                                "azure-core (<2.0.0,>=1.8.0)",
+                                "azure-mgmt-core (<2.0.0,>=1.2.0)",
+                                "marshmallow (<4.0.0,>=3.5)",
+                                "tqdm",
+                                "azure-storage-blob (<=12.5.0,>12.0.0b4)",
+                                "pydash (<=4.9.0)",
+                                "azure-storage-file-share (==12.3.0)",
+                                "pathspec (==0.8.*)",
+                                "cryptography (<=3.3.2)"
```
Comment (fengzhou-msft) [inline src/index.json:10347]: Can you remove `pyyaml`, `msrest`, `azure-core` and `azure-mgmt-core`? They're already in the dependency list of `azure-cli`. 

It is also recommended to replace `azure-storage-blob` and `azure-storage-file-share` with `azure-multiapi-storage` installed by `azure-cli`. You can refer to this PR: https://github.com/Azure/azure-cli/pull/13414. @Juliehzl can help if you have any questions about storage dependency usages.

Comment (Mossaka) [reply src/index.json]: I removed these dependencies in index.json. Do I also need to remove them in `setup.py`? Will ping @Juliehzl for azure-storage-blob dependency.

Comment (fengzhou-msft) [reply src/index.json]: You need to remove them in setup.py and regenerate the wheel and update index.json.

Comment (Juliehzl) [reply src/index.json]: After discussed with @Mossaka, the storage sdk is required by azure-ml package, not cli extension. If there is no packaging issue, I am fine with it.

Comment (Mossaka) [reply src/index.json]: @Juliehzl LMK if there is no packaging issue.

Comment (Mossaka) [reply src/index.json]: Removed dependencies in the CLI wheel @fengzhou-msft

**Comment by fengzhou-msft** [comment]
Merge https://github.com/Azure/azure-cli-extensions/pull/3317 first before merging this PR.

**Pull request closed without merging**